spec.html

```diff
@@ -7890,12 +7890,25 @@ <h1>FunctionDeclarationInstantiation ( _func_, _argumentsList_ )</h1>
-    <p>Unless otherwise specified every built-in function object has the %FunctionPrototype% object as the initial value of its [[Prototype]] internal slot.</p>
+    <p>Every built-in function object has a [[Realm]] internal slot whose value is the Realm Record of the realm for which the object was initially created. Unless otherwise specified, every built-in function object has the %FunctionPrototype% object as the initial value of its [[Prototype]] internal slot.</p>
     <p>The behaviour specified for each built-in function via algorithm steps or other means is the specification of the function body behaviour for both [[Call]] and [[Construct]] invocations of the function. However, [[Construct]] invocation is not supported by all built-in functions. For each built-in function, when invoked with [[Call]], the [[Call]] _thisArgument_ provides the *this* value, the [[Call]] _argumentsList_ provides the named parameters, and the NewTarget value is *undefined*. When invoked with [[Construct]], the *this* value is uninitialized, the [[Construct]] _argumentsList_ provides the named parameters, and the [[Construct]] _newTarget_ parameter provides the NewTarget value. If the built-in function is implemented as an ECMAScript function object then this specified behaviour must be implemented by the ECMAScript code that is the body of the function. Built-in functions that are ECMAScript function objects must be strict functions. If a built-in constructor has any [[Call]] behaviour other than throwing a *TypeError* exception, an ECMAScript implementation of the function must be done in a manner that does not cause the function's [[FunctionKind]] internal slot to have the value `"classConstructor"`.</p>
+  <p>Unless otherwise specified in the description of a particular function, if a built-in function or constructor is given fewer arguments than the function is specified to require, the function or constructor shall behave exactly as if it had been given sufficient additional arguments, each such argument being the *undefined* value. Such missing arguments are considered to be &ldquo;not present&rdquo; and may be identified in that manner by specification algorithms.</p>
+    <p>Unless otherwise specified in the description of a particular function, if a built-in function or constructor described is given more arguments than the function is specified to allow, the extra arguments are evaluated by the call and then ignored by the function. However, an implementation may define implementation specific behaviour relating to such arguments as long as the behaviour is not the throwing of a *TypeError* exception that is predicated simply on the presence of an extra argument.</p>
```

Comment:
I’m not sure about this last part - why should an implementation be forbidden one of the API features userland libraries have (to throw when a function is provided with the wrong number of arguments)?

The only thing i can think of is to preserve our ability to add future arguments - if this is the reason, i think that should be called out here.

Reply:
To be honest, prohibiting TypeErrors in response to excess arguments seems _hostile_ to a desire for adding more arguments in the future, because it cuts off a convenient path for user code to test the execution environment. But regardless, I don't want to change the significance of such text in a simple "rearrange" PR like this, though I would contemplate doing so in a followup if you like.

Reply:
This was an intentional change made in ES5 for interoperability reasons.  ES3 had said that implementations were permitted (but not required) to throw in such cases.  That was non-interoperable because code written for an implementation that did not throw in such cases would fail on an implementation that did throw.  There are really only two interoperable alternatives either require all implementations to throw or require that no implementations throw.  Requiring a throw would have been a much bigger departure from the then (and still) current practice.